checkov/terraform/checks/resource/gcp/GoogleComputeSerialPorts.py

```diff
@@ -1,21 +1,20 @@
-from checkov.terraform.checks.resource.base_resource_check import BaseResourceCheck
-from checkov.common.models.enums import CheckResult, CheckCategories
+from checkov.terraform.checks.resource.base_resource_negative_value_check import BaseResourceNegativeValueCheck
+from checkov.common.models.enums import CheckCategories
 
 
-class GoogleComputeSerialPorts(BaseResourceCheck):
+class GoogleComputeSerialPorts(BaseResourceNegativeValueCheck):
     def __init__(self):
         name = "Ensure 'Enable connecting to serial ports' is not enabled for VM Instance"
         id = "CKV_GCP_35"
         supported_resources = ['google_compute_instance']
         categories = [CheckCategories.NETWORKING]
         super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
 
-    def scan_resource_conf(self, conf):
-        if 'metadata' in conf.keys():
-            if 'serial-port-enable'in conf['metadata'][0]:
-                if conf['metadata'][0]['serial-port-enable'] not in ['0', False]:
-                    return CheckResult.FAILED
-        return CheckResult.PASSED
+    def get_inspected_key(self):
+        return 'metadata/[0]/serial-port-enable'
+
+    def get_forbidden_values(self):
+        return ['1', True]
```

Comment:
@tronxd Is this required? Or does chekov parse `1` and `"true"` as `true`?

Reply:
@nimrodkor yes, as a backward compatibility feature for Terraform v0.11 syntax
refer the function `evaluate_string_booleans` at `checkov/terraform/runner.py`

Reply:
@tronxd maybe we should maintain that list in a common enum?

Reply:
@schosterbarak @rotemavni scanning occurs after `1` would be evaluated into `True`, so the forbidden values should be `[True]` only